### PR TITLE
Bresenham's ALgorithm Implementation

### PR DESCRIPTION
**Pull request by ImranNawar**
### Describe your change:

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: graphics/midPointAlgorihtm.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Draw a line between two points using Mid point algorithm (Bresenham's Agorithm)
+"""
+import sys
+
+from OpenGL.GL import *
+from OpenGL.GLU import *
+from OpenGL.GLUT import *
+
+
+def init():
```
[the remark this answers is not in the document]
Comment (ImranNawar) [reply graphics/midPointAlgorihtm.py:11]: The function does not return anything. It's just draw a line  as the program run.

**Comment by ImranNawar** [comment]
The function in program does not return anything. It's just draw a line  as the program run.

File: graphics/digital-differential-analyzer.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+from OpenGL.GLUT import *
+
+
+def init():
+    glClearColor(0.0, 1.0, 0.0, 1.0)
+    gluOrtho2D(-100.0, 100.0, -100.0, 100.0)
+
+
+# Function for digital differential analyzer algorithm
+def draw_line_dda(x1, y1, x2, y2):
```
Comment (cclauss) [inline graphics/digital-differential-analyzer.py:17]:
```suggestion
def draw_line_dda(x1: float, y1: float, x2: float, y2: float) -> None:
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: graphics/midPointAlgorihtm.py (hunk 1 above)
Comment (cclauss) [inline graphics/midPointAlgorihtm.py:11]:
```suggestion
def init() -> None:
```

File: graphics/digital-differential-analyzer.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+def draw_line_dda(x1: float, y1: float, x2: float, y2: float) -> None:
+    dx = x2 - x1
+    dy = y2 - y1
+    steps = abs(dx) if abs(dx) > abs(dy) else abs(dy)
+    x_inc = dx / steps
+    y_inc = dy / steps
+    x = x1
+    y = y1
+    glBegin(GL_POINTS)
+    for i in range(int(steps)):
```
Comment (cclauss) [inline graphics/digital-differential-analyzer.py:26]:
```suggestion
    for _ in range(int(steps)):
```

File: graphics/digital-differential-analyzer.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
+"""
+Draws a line between two given points using the DDA algorithm.
+"""
+import sys
+
+from OpenGL.GL import *
```
Comment (cclauss) [inline graphics/digital-differential-analyzer.py:6]:
```suggestion
from OpenGL.GL import glVertex2f
```

**Pull request closed without merging**